### ISSUE TITLE
Invalid JSON - metrics-addr

### ISSUE THREAD

**Issue by alexellis** [docker/for-mac#1927]
### Expected behavior

Setting metrics-addr in settings panel enables Prometheus metrics.

As per docs:

https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-metrics

https://docs.docker.com/engine/admin/prometheus/#configure-and-run-prometheus

### Actual behavior

"invalid json / unsupported option"

<img width="375" alt="screen shot 2017-08-02 at 08 53 45" src="https://user-images.githubusercontent.com/6358735/28863544-3deeb3cc-7761-11e7-99a3-3983605020de.png">

### Information

```
Client:
 Version:      17.07.0-ce-rc1
 API version:  1.31
 Go version:   go1.8.3
 Git commit:   8c4be39
 Built:        Wed Jul 26 05:20:09 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.07.0-ce-rc1
 API version:  1.31 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   8c4be39
 Built:        Wed Jul 26 05:25:01 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Comment by thaJeztah** [comment]
ping @ebriney PTAL

confirmed that this option is missing in the validator script

**Comment by ebriney** [comment]
thanks i'll will add it to the scheme file here
https://github.com/docker/pinata/blob/master/common/resources/linux-daemon-options.json
is there a way to get it updated automatically from moby/moby Sebastiaan?

**Comment by thaJeztah** [comment]
@ebriney not easily, but it's a general problem that we should look into (see https://github.com/moby/moby/pull/34093#issuecomment-316698133)

**Comment by thaJeztah** [comment]
For now, would it be an option to just _warn_ about an (possibly) invalid option? That would resolve the case where someone made a typo, but put the responsibility with the user.

**Comment by ebriney** [comment]
For sure we can do that.
Although it is still possible to edit the file without the gui.
It is in `~/.docker/daemon.json`

**Comment by thaJeztah** [comment]
> It is in `~/.docker/daemon.json`

oh, wow, I didn't know D4M stored it there nowadays 👍 should D4M also store it in `/etc/docker/daemon.json` to be consistent with Linux?

**Comment by Serjk** [comment]
I have a similar issue with setting "bip" option. 

<img width="378" alt="screen shot 2017-08-03 at 18 37 42" src="https://user-images.githubusercontent.com/1616504/28930268-1b57f786-787b-11e7-8b32-3dd0b6f86705.png">

_Docker version 17.07.0-ce-rc1, build 8c4be39_

**Comment by ManoMarks** [comment]
I ran into the same issue, then upgraded to Version 17.07.0-ce-rc2-mac22 (18909). Now there's a new issue "metrics-addr" : Type not valid
![screen shot 2017-08-11 at 2 41 55 pm](https://user-images.githubusercontent.com/113855/29233028-abba8b1c-7ea3-11e7-89cc-f139d6136410.png)

**Comment by gtardif** [comment]
This has been fixed in 17.07.0-ce-rc4. Please let us know if you still have issues with the latest version.

**Comment by Serjk** [comment]
It works as expected on the latest version. Thanks.
